device/tests/Microsoft.Azure.Devices.Client.Test/ModuleAuthenticationWithTokenRefreshTests.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information
+
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Diagnostics;
+using Microsoft.Azure.Amqp;
+
+namespace Microsoft.Azure.Devices.Client.Test
+{
+    [TestClass]
+    public class ModuleAuthenticationWithTokenRefreshTests
+    {
+        private const string TestDeviceId = "TestDeviceID";
+        private const string TestModuleId = "TestModuleID";
+        private const string TestIoTHubName = "contoso.azure-devices.net";
+        private const int DefaultTimeToLiveSeconds = 1 * 60 * 60;
+        private static string TestSharedAccessKey;
+
+        static ModuleAuthenticationWithTokenRefreshTests()
+        {
+            var rnd = new Random();
+            var rndBytes = new byte[32];
+            rnd.NextBytes(rndBytes);
+
+            TestSharedAccessKey = Convert.ToBase64String(rndBytes);
+        }
+
+        [TestMethod]
+        public void ModuleAuthenticationWithTokenRefresh_Ctor_WrongArguments_Fail()
+        {
+            TestAssert.Throws<ArgumentNullException>(() => new TestImplementation(null, TestModuleId));
+            TestAssert.Throws<ArgumentNullException>(() => new TestImplementation(TestDeviceId, null));
+            TestAssert.Throws<ArgumentNullException>(() => new TestImplementation("   ", TestModuleId));
+            TestAssert.Throws<ArgumentNullException>(() => new TestImplementation(TestDeviceId, "  "));
+            TestAssert.Throws<ArgumentOutOfRangeException>(() => new TestImplementation(TestDeviceId, TestModuleId, -1, 10));
+            TestAssert.Throws<ArgumentOutOfRangeException>(() => new TestImplementation(TestDeviceId, TestModuleId, 60, -1));
+            TestAssert.Throws<ArgumentOutOfRangeException>(() => new TestImplementation(TestDeviceId, TestModuleId, 60, 101));
+        }
+
+        [TestMethod]
+        public void ModuleAuthenticationWithTokenRefresh_Ctor_DefaultsGetProperties_Ok()
```

Comment:
This is already tested in the base class UnitTest. If you want to, you can test the particular behavior for Module (i.e. token is created with the module name, etc)
Otherwise I think you can delete these duplicates as you've deleted the Device ones.

Reply:
I thought in this one case, the some duplicate test code is okay. Want to have a test that tests it for the derived type.